### PR TITLE
Use native Object.assign if available

### PR DESCRIPTION
**Pull request by asaf-romano**
My fault. It turns out that the polyfill isn’t nearly as fast as the native implementation (at least in Firefox).

### CODE REVIEW

File: src/internal/_objectAssign.js [hunk 1]
```diff
@@ -1,24 +1,28 @@
 var _has = require('./_has');
 
-// Based on https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
-module.exports = function _objectAssign(target) {
-  if (target == null) {
-    throw new TypeError('Cannot convert undefined or null to object');
-  }
+module.exports = (function() {
+  return typeof Object.assign === 'function' ?
```
Comment (buzzdecafe) [inline src/internal/_objectAssign.js]: does it make sense to do this conditional logic in its own module and then use `internal/_assign` instead?

Comment (asaf-romano) [inline src/internal/_objectAssign.js]: I followed `keys` here. I cannot think of a use case for the standalone module, but it if it helps readability it's probably worth it anyway.

Comment (jdalton) [inline src/internal/_objectAssign.js]: Builtin `Object.assign` will also copy symbols over.

**Comment by CrossEye** [comment]
@jdalton:

> Builtin `Object.assign` will also copy symbols over.

That's an interesting point.  We've been mostly ignoring Symbols.  It's hard to come to terms with them from an FP perspective.  At some point we're going to have to figure out what we want to do, though.

**Comment by jdalton** [comment]
While implementation perf of `Object.assign` is improving its baggage has kept me from using it.

**Comment by CrossEye** [comment]
:herb:

File: src/internal/_assign.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+var _objectAssign = require('./_objectAssign');
+
+module.exports = (function() {
+  return typeof Object.assign === 'function' ? Object.assign : _objectAssign;
+}());
```
Comment (davidchambers) [inline src/internal/_assign.js]: Is the IIFE necessary?

Comment (asaf-romano) [inline src/internal/_assign.js]: Nope.

**Pull request merged**